ultra/train.py

```diff
@@ -0,0 +1,182 @@
+                **eval_info,
+                **env.info,
+            )
+            action = agent.act(state, explore=True)
+            observations, rewards, dones, infos = env.step({AGENT_ID: action})
+            next_state = observations[AGENT_ID]
+
+            loss_output = agent.step(
+                state=state,
+                action=action,
+                reward=rewards[AGENT_ID],
+                next_state=next_state,
+                done=dones[AGENT_ID],
+            )
+            episode.record_step(
+                agent_id=AGENT_ID,
+                infos=infos,
+                rewards=rewards,
+                total_step=total_step,
+                loss_output=loss_output,
+            )
+            total_step += 1
+            state = next_state
+
+        episode.record_episode()
+        episode.record_tensorboard(agent_id=AGENT_ID)
+        if finished:
+            break
+
+    env.close()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser("intersection-single-agent")
+    parser.add_argument(
+        "--task", help="Tasks available : [1, 2]", type=str, default="1"
+    )
+    parser.add_argument(
+        "--level",
+        help="Levels available : [easy, medium, hard, no-traffic]",
+        type=str,
+        default="easy",
+    )
+    parser.add_argument(
+        "--episodes", help="Number of training episodes", type=int, default=1000000
+    )
+    parser.add_argument(
+        "--timestep", help="Environment timestep (sec)", type=float, default=0.1
+    )
+    parser.add_argument(
+        "--headless", help="Run without envision", type=bool, default=False
+    )
+    parser.add_argument(
+        "--eval-episodes", help="Number of evaluation episodes", type=int, default=200
+    )
+    parser.add_argument(
+        "--eval-rate",
+        help="Evaluation rate based on number of observations",
+        type=int,
+        default=10000,
+    )
+
+    parser.add_argument(
+        "--seed",
+        help="Environment seed",
+        default=2,
+        type=int,
+    )
+    args = parser.parse_args()
+
+    num_cpus = max(
+        1, psutil.cpu_count(logical=False) - 1
+    )  # remove `logical=False` to use all cpus
+
+    policy_class = "ultra.baselines.sac:sac-v0"
+    # ray_kwargs = default_ray_kwargs(num_cpus=num_cpus, num_gpus=num_gpus)
+    ray.init()  # **ray_kwargs)
+    # try:
+    ray.wait(
+        [
+            train.remote(
+                scenario_info=(args.task, args.level),
+                num_episodes=int(args.episodes),
+                eval_info={
+                    "eval_rate": float(args.eval_rate),
+                    "eval_episodes": int(args.eval_episodes),
+                },
+                timestep_sec=float(args.timestep),
+                headless=args.headless,
+                policy_class=policy_class,
+                seed=args.seed,
+            )
+        ]
+    )
+    # finally:
+    #     time.sleep(1)
+    #     ray.shutdown()
```

Comment:
Sorry for the late review. Everything seems good, train.py will be updated from my side